modin/core/io/column_stores/column_store_dispatcher.py

```diff
@@ -158,16 +160,20 @@ def build_columns(cls, columns):
             List with lengths of `col_partitions` subarrays
             (number of columns that should be read by workers).
         """
-        num_partitions = NPartitions.get()
-        column_splits = (
-            len(columns) // num_partitions
-            if len(columns) % num_partitions == 0
-            else len(columns) // num_partitions + 1
-        )
-        col_partitions = [
-            columns[i : i + column_splits]
-            for i in range(0, len(columns), column_splits)
-        ]
+        columns_length = len(columns)
+        if columns_length == 0:
+            col_partitions = []
+        else:
+            num_partitions = NPartitions.get()
+            column_splits = (
+                columns_length // num_partitions
+                if columns_length % num_partitions == 0
+                else columns_length // num_partitions + 1
+            )
+            col_partitions = [
```

Comment:
This is potentially more trouble than its worth, but we could do this:
```python
col_partitions, column_widths = zip(*[
                 (columns[i : i + column_splits], len(columns[i : i + column_splits]))
                 for i in range(0, len(columns), column_splits)
])
return list(col_partitions), list(column_widths])
```
but this makes the code slightly less readable to save an additional map over the columns which idk if its worth it.

Reply:
I wouldn't expect that to improve performance much. Both versions do the same map to get the column splits, and the step of computing lengths as we do now should be fairly fast as it's just a map over a list of size `NPartitions`. `zip` has to go through all of the new splits, too.

Anyway, this is unrelated to this PR, but let me know offline if you want to talk about it more.